src/components/Api.js

```diff
@@ -0,0 +1,150 @@
+export default class Api {
+  constructor({ baseUrl, headers: { authorization } }) {
+    this.baseUrl = baseUrl;
+    this.headers = {
+      authorization,
+    };
+  }
+  getUserData() {
+    return fetch(`${this.baseUrl}/users/me`, {
+      headers: {
+        authorization: this.headers.authorization,
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+  updateProfileInfo(name, description) {
+    return fetch(`${this.baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        authorization: this.headers.authorization,
+        "Content-Type": "application/json",
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://disk.yandex.ru/i/J93dbJp7eypXKQ . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)